Source/monster.cpp

```diff
@@ -3334,7 +3349,7 @@ void MAI_Round(int i, bool special)
 			int dist = std::max(abs(mx), abs(my));
 			if ((Monst->_mgoalvar1++ >= 2 * dist && DirOK(i, md)) || dTransVal[Monst->position.tile.x][Monst->position.tile.y] != dTransVal[fx][fy]) {
 				Monst->_mgoal = MGOAL_NORMAL;
-			} else if (!M_RoundWalk(i, md, &Monst->_mgoalvar2)) {
+			} else if (!M_RoundWalk(i, md, (RotationDirection *)&Monst->_mgoalvar2)) {

```

Comment:
If we're casting pointers, shouldn't we specify the underlying type of the enum?

Reply:
`_mgoalvar2` os one of those nasty temp variables that are used as different things in various contexts. Some times is a relative direciton (`left`/`right`) some times it's a cardinal `Direciton`, some times it's the `X` component of a position.

Reply:
Perhaps it was unclear what I was suggesting because of the line that I chose to comment on. Since the type of `_mgoalvar2` is `int`, I think the underlying type of the `RotationDirection` enum should also be `int`. From what I'm reading, unless we explicitly specify the type, the standard only specifies that the underlying type will be big enough to represent all its values.

```C++
enum RotationDirection : int {
	RotateRight,
	RotateLeft
};
```

Reply:
Yeah, this is undefined behaviour territory.

Given it's a new enum that isn't used anywhere else I'd really like to use scoped enums (`enum class|struct`) instead. They default to a backing type of int which avoids this issue.